Source/misdat.cpp

```diff
@@ -128,113 +128,152 @@ MissileData MissilesData[] = {
 	// clang-format on
 };
 
+namespace {
+
+constexpr std::array<uint8_t, 16> Repeat(uint8_t v) // NOLINT(readability-identifier-length)
+{
+	return { v, v, v, v, v, v, v, v, v, v, v, v, v, v, v, v };
+}
+
+const std::array<uint8_t, 16> MissileAnimDelays[] {
+	{},
+	Repeat(1),
+	Repeat(2),
+	{ 0, 1 },
+	{ 1 },
+};
+
+const std::array<uint8_t, 16> MissileAnimLengths[] {
+	{},
+	Repeat(1),
+	Repeat(4),
+	Repeat(5),
+	Repeat(6),
+	Repeat(7),
+	Repeat(8),
+	Repeat(9),
+	Repeat(10),
+	Repeat(12),
+	Repeat(13),
+	Repeat(14),
+	Repeat(15),
+	Repeat(16),
+	Repeat(17),
+	Repeat(19),
+	Repeat(20),
+	{ 9, 4 },
+	{ 15, 4, 3 },
+	{ 13, 11 },
+	{ 16, 16, 16, 16, 16, 16, 16, 16, 8 }

```

Comment:
it should also be possible to split the sprites for these missiles to avoid the special cases